Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Sandboxed/SiteStateListenerScriptHandler.swift

```diff
@@ -61,57 +61,57 @@ class SiteStateListenerScriptHandler: TabContentScript {
         return
       }
       
-      if let pageData = tab.currentPageData, let frameInfo = pageData.framesInfo[frameURL] {
-        let domain = pageData.domain(persistent: !tab.isPrivate)
-        let models = AdBlockStats.shared.cosmeticFilterModels(forFrameURL: frameURL, domain: domain)
-        
-        let hideSelectors = models.reduce(Set<String>(), { partialResult, model in
-          return partialResult.union(model.hideSelectors)
-        })
-        
-        var styleSelectors: [String: Set<String>] = [:]
-        
-        for model in models {
-          for (key, values) in model.styleSelectors {
-            styleSelectors[key] = styleSelectors[key]?.union(Set(values)) ?? Set(values)
+      if let pageData = tab.currentPageData {
+        Task { @MainActor in
+          let domain = pageData.domain(persistent: !tab.isPrivate)
+          let models = await AdBlockStats.shared.cosmeticFilterModels(forFrameURL: frameURL, domain: domain)
+          
+          let hideSelectors = models.reduce(Set<String>(), { partialResult, model in
+            return partialResult.union(model.hideSelectors)
+          })
+          
+          var styleSelectors: [String: Set<String>] = [:]
+          
+          for model in models {
+            for (key, values) in model.styleSelectors {
+              styleSelectors[key] = styleSelectors[key]?.union(Set(values)) ?? Set(values)
+            }
           }
-        }
-        
-        let styleSelectorObjects = styleSelectors.map { selector, rules -> UserScriptType.SelectorsPollerSetup.StyleSelectorEntry in
-          UserScriptType.SelectorsPollerSetup.StyleSelectorEntry(
-            selector: selector, rules: rules
+          
+          let styleSelectorObjects = styleSelectors.map { selector, rules -> UserScriptType.SelectorsPollerSetup.StyleSelectorEntry in
+            UserScriptType.SelectorsPollerSetup.StyleSelectorEntry(
+              selector: selector, rules: rules
+            )
+          }
+          
+          let setup = UserScriptType.SelectorsPollerSetup(
+            frameURL: frameURL,
+            genericHide: models.contains { $0.genericHide },
+            hideSelectors: hideSelectors,
+            styleSelectors: Set(styleSelectorObjects)
+          )
+          
+          let encoder = JSONEncoder()
+          let data = try encoder.encode(setup)
```

Comment:
How large of a data set could this be? Perhaps we could execute this off-main

Reply:
There may be a few hundred selectors here so not too large but happy to move this off the main thread.

Reply:
Let me know what you think of this approach: https://github.com/brave/brave-ios/pull/6507/commits/b6efca30fc4df0cea5de35a9a043a1386e03b4a3